airflow/contrib/hooks/kubernetes_hook.py

```diff
@@ -0,0 +1,53 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import tempfile
+
+from kubernetes import client, config
+
+from airflow.hooks.base_hook import BaseHook
+
+
+class Kuberneteshook(BaseHook):
+    """
+    creates kubernetes API connection.
+    :param conn_id: the connection to Kubernetes cluster
+    """
+
+    def __init__(
+        self,
+        conn_id="kubernetes_default"
+    ):
+        self.connection = self.get_connection(conn_id)
+
+    def get_conn(self):
+        """
+        Returns kubernetes api session for use with requests
+        """
+        if self.connection.extra is None or self.connection.extra == '':
+            self.log.info("loading kube_config from: default file")
+        else:
+            extra_obj = self.connection.extra_dejson
+            if "in_cluster" in extra_obj.keys():
+                self.log.info("loading kube_config from: in_cluster configuration")
+                config.load_incluster_config()
+            else:
+                with tempfile.NamedTemporaryFile() as temp_config:
+                    self.log.info("loading kube_config from: connection extra json")
+                    temp_config.write(self.connection.extra.encode())
```

Comment:
The configuration must be saved under the dictionary key. You can't pass the entire dictionary.